spec/requests/api/v1/keywords_controller_spec.rb

```diff
@@ -144,4 +144,64 @@
       end
     end
   end
+
+  describe 'GET#show' do
+    context 'when CSV file is valid' do
+      it 'returns success status' do
+        keyword = Fabricate(:keyword_parsed)
+        user = keyword.user
+        get api_v1_keyword_path(keyword.id), headers: create_token_header(user)
+
+        expect(response).to have_http_status(:success)
+      end
+
+      it 'returns same ads_page_count as the keyword' do
+        keyword = Fabricate(:keyword_parsed)
+        user = keyword.user
+        get api_v1_keyword_path(keyword.id), headers: create_token_header(user)
+
+        expect(JSON.parse(response.body)['data']['attributes']['ads_page_count']).to eq(keyword.reload.ads_page_count)
+      end
+
+      it 'returns json with source as an included relationship' do
+        keyword = Fabricate(:keyword_parsed)
+        user = keyword.user
+        get api_v1_keyword_path(keyword.id), headers: create_token_header(user)
+
+        expect(JSON.parse(response.body)['included'].find { |included| included['type'] == 'source' }['attributes']['name']).to eq(keyword.reload.source.name)
+      end
+    end
+
+    context 'when keyword does not belonged to the user' do
+      it 'returns not found status' do
+        keyword = Fabricate(:keyword_parsed)
+        get api_v1_keyword_path(keyword.id), headers: create_token_header
+
+        expect(response).to have_http_status(:not_found)
+      end
+
+      it 'returns the api.errors.not_found error' do
+        keyword = Fabricate(:keyword_parsed)
+        get api_v1_keyword_path(keyword.id), headers: create_token_header
+
+        expect(JSON.parse(response.body)['errors']['details']).to include(I18n.t('api.errors.not_found'))
+      end
+    end
+
+    context 'when there is no keyword' do
+      it 'returns not found status' do
+        get api_v1_keyword_path(0), headers: create_token_header
+
+        expect(response).to have_http_status(:not_found)
+      end
+    end
+
+    context 'given a non-logged in user' do
```

Comment:
```suggestion
    context 'given no token headers' do
```
I do not see any "unlogged user" in the test below 🙈